lightkurve/correctors/pldcorrector.py

```diff
@@ -139,144 +237,284 @@ def correct(self, aperture_mask=None, cadence_mask=None, gp_timescale=30,
+    """Correct Kepler light curves by detrending against local pixel time series.
+
+    Subclass of `.PLDCorrector`, a version of the `.RegressionCorrector` class
+    in which the `.DesignMatrix` is constructed from pixel time series.
+
+    Returns a systematics-corrected light curve.
+
+    Parameters
+    ----------
+    dm : `.DesignMatrix` or `.DesignMatrixCollection` or `None`
+        One or more design matrices.  Each matrix must have a shape of
+        (time, regressors). The columns contained in each matrix must be
+        known to correlate with additive noise components we want to remove
+        from the light curve. If `None`, a `.DesignMatrix` will be generated
+        using the other keyword arguments.
+    pld_order : int
+        The order of Pixel Level De-correlation to be performed. First order
+        (`n=1`) uses only the pixel fluxes to construct the design matrix.
+        Higher order populates the design matrix with columns constructed
+        from the products of pixel fluxes.
+    pixel_components : int
+        Number of principal components derived from the background pixel
+        time series to utilize.
+    background_mask : array-like or None
+        A boolean array flagging the background pixels such that `True` means
+        that the pixel will be used to generate the background systematics model.
+        If `None`, all pixels which are fainter than 1-sigma above the median
+        flux will be used.
+    pld_aperture_mask : array-like, 'pipeline', 'all', 'threshold', or None
+        A boolean array describing the aperture such that `True` means
+        that the pixel will be used when selecting the PLD basis vectors.
+        If `None` or `all` are passed in, all pixels will be used.
+        If 'pipeline' is passed, the mask suggested by the official pipeline
+        will be returned.
+        If 'threshold' is passed, all pixels brighter than 3-sigma above
+        the median flux will be used.
+    spline_n_knots : int
+        Number of knots in spline.
+    spline_degree : int
+        Polynomial degree of spline.
+    n_pca_terms : int
+        Number of terms added to the design matrix from each order of PLD
+        when performing Principal Component Analysis for models higher than
+        first order. Increasing this value may provide higher precision at
+        the expense of computational time.
+    restore_trend : bool
+        Whether to restore the long term spline trend to the light curve.
+    sparse : bool
+        Whether to create `SparseDesignMatrix`.
+    **kwargs : dict
+        Extra parameters to be passed to `RegressionCorrector.correct`.
+
+    Returns
+    -------
+    clc : `.KeplerLightCurve`
+        Noise-corrected `.KeplerLightCurve`.
+    """
+
+    def __repr__(self):
+        return 'KeplerPLDCorrector (LC: {})'.format(self.lc.label)
+
+    def correct(self, dm=None, pld_order=3, pixel_components=15, background_mask=None,
```

Comment:
Can you add a few words to the docstring why `pld_order` and `pixel_components` have a different default for Kepler?  Are we certain that this is necessary?  Are these defaults not suitable for TESS?  Could there be a way to determine the defaults based on generic properties (e.g. amount of motion, number of pixels in TPF, number of cadences, ...) rather than mission name?

I ask these questions because it is way more expensive to maintain both `KeplerPLDCorrector` and `TessPLDCorrector` classes compared to a single `PLDCorrector` class, so we want to be absolutely certain that it is necessary to introduce this class hierarchy!